MapboxCoreNavigation/RouteController.swift

```diff
@@ -134,6 +134,16 @@ public protocol RouteControllerDelegate: class {
     */
     @objc(routeController:didArriveAtWaypoint:)
     optional func routeController(_ routeController: RouteController, didArriveAt waypoint: Waypoint) -> Bool
+    
+    /**
+     Called when a tunnel is detected on a route step.
+     
+     Implement this method to enable dark mode when a commuter enters a tunnel.
+     
+     - parameter darkModeEnabled: The darkModeEnabled flag determines whether a `DarkStyle` should apply.
+     */
+    @objc(didEnterTunnel:)
+    optional func didEnterTunnel(_ darkModeEnabled: Bool)
```

Comment:
We should remove the notion of dark mode from `RouteController` because MapboxCoreNavigation deals with non-UI related functionality.

Reply:
Makes sense. Will do.